Name: Tabios, Gavrie Adrian W.
Student Number: 2022103939
Course/Section: DS100-1/B31
M1 Programming Exercise

In [ ]:
pip install requests

In [ ]:
pip install pandas

In [ ]:
pip install openpyxl

In [4]:
import requests as rq
from io import BytesIO

In [5]:
import numpy as np
import pandas as pd

In [7]:
import requests as rq 
from io import StringIO
import pandas as pd

url = 'https://raw.githubusercontent.com/ABRION77/DS-100-repository/main/WHR_2019.csv'
response = rq.get(url)
data = pd.read_csv(StringIO(response.text))
pd.set_option('display.max_rows', None)
data.head()

Overall rank Country or region  Score  GDP per capita  Social support  \
0             1           Finland  7.769           1.340           1.587   
1             2           Denmark  7.600           1.383           1.573   
2             3            Norway  7.554           1.488           1.582   
3             4           Iceland  7.494           1.380           1.624   
4             5       Netherlands  7.488           1.396           1.522   

   Healthy life expectancy  Freedom to make life choices  Generosity  \
0                    0.986                         0.596       0.153   
1                    0.996                         0.592       0.252   
2                    1.028                         0.603       0.271   
3                    1.026                         0.591       0.354   
4                    0.999                         0.557       0.322   

   Perceptions of corruption  
0                      0.393  
1                      0.410  
2                      0.341  
3                      0.118  
4                      0.298

In [10]:
# Assume 'Overall rank' column that contains the numerival values from 1 - 69
happiness_data = data[data['Overall rank'].between(1,69)]

# This line of code is not really necessary as there was no NaN Values in all iterations and columns on the csv file
happiness_data.update(happiness_data[['Score', 'GDP per capita']].fillna(0))
happiness_data

Overall rank     Country or region  Score  GDP per capita  Social support  \
0              1               Finland  7.769           1.340           1.587   
1              2               Denmark  7.600           1.383           1.573   
2              3                Norway  7.554           1.488           1.582   
3              4               Iceland  7.494           1.380           1.624   
4              5           Netherlands  7.488           1.396           1.522   
5              6           Switzerland  7.480           1.452           1.526   
6              7                Sweden  7.343           1.387           1.487   
7              8           New Zealand  7.307           1.303           1.557   
8              9                Canada  7.278           1.365           1.505   
9             10               Austria  7.246           1.376           1.475   
10            11             Australia  7.228           1.372           1.548   
11            12            Costa Rica  7.167           1.034           1.441   
12            13                Israel  7.139           1.276           1.455   
13            14            Luxembourg  7.090           1.609           1.479   
14            15        United Kingdom  7.054           1.333           1.538   
15            16               Ireland  7.021           1.499           1.553   
16            17               Germany  6.985           1.373           1.454   
17            18               Belgium  6.923           1.356           1.504   
18            19         United States  6.892           1.433           1.457   
19            20        Czech Republic  6.852           1.269           1.487   
20            21  United Arab Emirates  6.825           1.503           1.310   
21            22                 Malta  6.726           1.300           1.520   
22            23                Mexico  6.595           1.070           1.323   
23            24                France  6.592           1.324           1.472   
24            25                Taiwan  6.446           1.368           1.430   
25            26                 Chile  6.444           1.159           1.369   
26            27             Guatemala  6.436           0.800           1.269   
27            28          Saudi Arabia  6.375           1.403           1.357   
28            29                 Qatar  6.374           1.684           1.313   
29            30                 Spain  6.354           1.286           1.484   
30            31                Panama  6.321           1.149           1.442   
31            32                Brazil  6.300           1.004           1.439   
32            33               Uruguay  6.293           1.124           1.465   
33            34             Singapore  6.262           1.572           1.463   
34            35           El Salvador  6.253           0.794           1.242   
35            36                 Italy  6.223           1.294           1.488   
36            37               Bahrain  6.199           1.362           1.368   
37            38              Slovakia  6.198           1.246           1.504   
38            39     Trinidad & Tobago  6.192           1.231           1.477   
39            40                Poland  6.182           1.206           1.438   
40            41            Uzbekistan  6.174           0.745           1.529   
41            42             Lithuania  6.149           1.238           1.515   
42            43              Colombia  6.125           0.985           1.410   
43            44              Slovenia  6.118           1.258           1.523   
44            45             Nicaragua  6.105           0.694           1.325   
45            46                Kosovo  6.100           0.882           1.232   
46            47             Argentina  6.086           1.092           1.432   
47            48               Romania  6.070           1.162           1.232   
48            49                Cyp

In [12]:
# Filtering based on GDP per capital this time
happiness_data = data[data["GDP per capita"].between(0.8, 1.5)]
print(happiness_data.head(24)) # This line of code only shows first few rows as a way to check

    Overall rank Country or region  Score  GDP per capita  Social support  \
0              1           Finland  7.769           1.340           1.587   
1              2           Denmark  7.600           1.383           1.573   
2              3            Norway  7.554           1.488           1.582   
3              4           Iceland  7.494           1.380           1.624   
4              5       Netherlands  7.488           1.396           1.522   
5              6       Switzerland  7.480           1.452           1.526   
6              7            Sweden  7.343           1.387           1.487   
7              8       New Zealand  7.307           1.303           1.557   
8              9            Canada  7.278           1.365           1.505   
9             10           Austria  7.246           1.376           1.475   
10            11         Australia  7.228           1.372           1.548   
11            12        Costa Rica  7.167           1.034           1.441   

In [13]:
happiness_data = data[data["GDP per capita"].between(0.8, 1.5)].copy()
happiness_data.loc[:, 'Generosity'] = np.round(happiness_data['GDP per capita'] / happiness_data['Social support'], 3)
happiness_data['Generosity'] = happiness_data['Generosity'].fillna(0)

In [15]:
# Filtering based on GDP per capita
happiness_data_filtered = data[data['GDP per capita'].between(0.8, 1.5)].copy()

# Calculating Generosity
happiness_data_filtered['Generosity'] = np.round(happiness_data_filtered['GDP per capita'] / happiness_data_filtered['Social support'], 3)
happiness_data_filtered['Generosity'] = happiness_data_filtered['Generosity'].fillna(0)

# Selecting required coluimns
happiness_data_filtered = happiness_data_filtered[['Overall rank', 'Country or region', 'GDP per capita', 'Healthy life expectancy', 'Generosity', 'Perceptions of corruption']]

# Display the first 25 rows for checking
print(happiness_data_filtered.head(24))

    Overall rank Country or region  GDP per capita  Healthy life expectancy  \
0              1           Finland           1.340                    0.986   
1              2           Denmark           1.383                    0.996   
2              3            Norway           1.488                    1.028   
3              4           Iceland           1.380                    1.026   
4              5       Netherlands           1.396                    0.999   
5              6       Switzerland           1.452                    1.052   
6              7            Sweden           1.387                    1.009   
7              8       New Zealand           1.303                    1.026   
8              9            Canada           1.365                    1.039   
9             10           Austria           1.376                    1.016   
10            11         Australia           1.372                    1.036   
11            12        Costa Rica           1.034  

In [20]:
# Calculation side of the programming exercises one statistical and the other for standard deviation
import numpy as np

# Filtering based on GDP per capita
happiness_data_filtered = data[data['GDP per capita'].between(0.8, 1.5)].copy()

# Equation 1: Calculating Generosity
happiness_data_filtered['Generosity'] = np.round(happiness_data_filtered['GDP per capita'] / happiness_data_filtered['Social support'], 3).fillna(0)

# Selecting Required Columns
happiness_data_filtered = happiness_data_filtered[['Overall rank', 'Country or region', 'GDP per capita', 'Healthy life expectancy', 'Generosity', 'Perceptions of corruption']]

# For checking display first few rows
print(happiness_data_filtered.head(24))

# Calculate statistics
stats_columns = ['Generosity', 'Perceptions of corruption', 'Healthy life expectancy']
statistics = {}

for column in stats_columns:
    mean = happiness_data_filtered[column].mean()
    median = happiness_data_filtered[column].median()
    mode = happiness_data_filtered[column].mode()[0]
    std = happiness_data_filtered[column].std()
    average = (mean + median + mode) / 3
    statistics[column] = {'Mean': mean, 'Median': median, 'Mode': mode, 'Average': average, 'Standard Deviation': std}

# Print/Display Statistics
for column, values in statistics.items():
    print(f"{column} statistics:")
    for stat, value in values.items():
        print(f"{stat}: {value}")
    print()


    Overall rank Country or region  GDP per capita  Healthy life expectancy  \
0              1           Finland           1.340                    0.986   
1              2           Denmark           1.383                    0.996   
2              3            Norway           1.488                    1.028   
3              4           Iceland           1.380                    1.026   
4              5       Netherlands           1.396                    0.999   
5              6       Switzerland           1.452                    1.052   
6              7            Sweden           1.387                    1.009   
7              8       New Zealand           1.303                    1.026   
8              9            Canada           1.365                    1.039   
9             10           Austria           1.376                    1.016   
10            11         Australia           1.372                    1.036   
11            12        Costa Rica           1.034  

In [23]:
summary_stats = happiness_data_filtered[['Country or region', 'GDP per capita', 'Healthy life expectancy', 'Generosity', 'Perceptions of corruption']].describe()
summary_stats.head(25)


GDP per capita  Healthy life expectancy  Generosity  \
count       97.000000                97.000000   97.000000   
mean         1.135134                 0.847876    0.848866   
std          0.194902                 0.164455    0.143109   
min          0.800000                 0.000000    0.562000   
25%          0.983000                 0.785000    0.754000   
50%          1.149000                 0.845000    0.847000   
75%          1.300000                 0.987000    0.921000   
max          1.500000                 1.122000    1.330000   

       Perceptions of corruption  
count                  97.000000  
mean                    0.110948  
std                     0.100368  
min                     0.004000  
25%                     0.038000  
50%                     0.080000  
75%                     0.142000  
max                     0.410000

Discuss the dataset that you have used for the programming exercise. Describe what kind of data processing you have made.

You may also want to discuss the significant features that can be observed in the dataset.

ANSWER:
The dataset I have choses is the World Happiness Report for 2019 from a website called Kaggle. The kind of data processing I have accomplished wanted to limit the data into three main columns namely GDP per capita, Healthy life expectancy, Generosity, and Perceptions of corruption. The mathematical equation conducted was simply focusing on mean, median, mode, average, and standard deviation.

The significant finding that I had on my data set is that the mean of the first tweny five countries reflected a high mean of 1.1351 which we can infer that having a hight GDP per capita aid's in increasing the overall happiness of their citizens, and the lowest affecting factor on happiness among the three is actually Perceptions of corruption with a low perception on corruption people become more happier knowing that their own government is not exploiting them as much or at least they perceived so.


Name: Tabios, Gavrie Adrian W.

Student Number: 2022103939
